premiere livaison (#1)

JanFLORENT opened the pull request.

JanFLORENT committed:
premiere livaison (#1)

pesublet commented on build.gradle:
```diff
@@ -0,0 +1,132 @@
+compileJava.options.encoding = 'UTF-8'
+
+ext {
+    paymentMethodApiVersion = '1.5'
```
Tu peux upgrader la version en 1.7

pesublet commented on build.gradle:
```diff
@@ -0,0 +1,132 @@
+
+plugins {
+    id "com.github.johnrengelman.shadow" version "5.1.0"
+    id "org.sonarqube" version "2.6.1"
```
Pour equens, on a upgradé à :
    id "org.sonarqube" version "2.8"
    id "org.owasp.dependencycheck" version "5.2.2"
Je pense que tu peux mettre ces versions

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/common/MerchantOrderReference.java:
```diff
@@ -0,0 +1,54 @@
+    @JacksonXmlProperty(localName = "merchandRef", namespace = "urn:PPEWShopServiceV3")
+    private String merchantRef;
+
+    public MerchantOrderReference() {
```
Constructeur à supprimer

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/response/PpewShopResponseKO.java:
```diff
@@ -0,0 +1,100 @@
+
+    public FailureCause getFailureCauseFromErrorCode() {
+        FailureCause cause;
+        switch (this.errorCode) {
```
petit null check avant

monext-ofrancome replied:
C'est vraiment résolu ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/exception/PluginException.java:
```diff
@@ -0,0 +1,103 @@
+     * @return A truncated errorCode to insert into any FailureResponse object.
+     */
+    public static String runtimeErrorCode(RuntimeException e) {
+        String errorCode = "plugin error: " + e.toString().substring(e.toString().lastIndexOf('.') + 1);
```
j'aimerais bien un TU et un petit commentaire car ce code n'est pas trivial

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationService.java:
```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public PaymentFormLogo getLogo(String paymentMethodIdentifier, Locale locale) {
+        String filename = config.get("logo.filename");
```
petit null check avec exception si pas de filename

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationService.java:
```diff
@@ -0,0 +1,74 @@
+    public PaymentFormLogo getLogo(String paymentMethodIdentifier, Locale locale) {
+        String filename = config.get("logo.filename");
+
+        InputStream input = this.getClass().getClassLoader().getResourceAsStream(filename);
```
fermeture du stream ou try with resources

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationService.java:
```diff
@@ -0,0 +1,74 @@
+            BufferedImage logo = ImageIO.read(input);
+
+            // Recover byte array from image
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
fermeture du stream ou try with resources

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationService.java:
```diff
@@ -0,0 +1,74 @@
+
+            // Recover byte array from image
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ImageIO.write(logo, config.get("logo.format"), baos);
```
null check

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImpl.java:
```diff
@@ -0,0 +1,177 @@
+    @Override
+    public ReleaseInformation getReleaseInformation() {
+        return ReleaseInformation.ReleaseBuilder.aRelease()
+                .withDate(LocalDate.parse(releaseProperties.get("release.date"), DateTimeFormatter.ofPattern("dd/MM/yyyy")))
```
null check

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+    public String getTransactionIdFromURL(String url) {
```
null check

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+    /**
+     * Define the methods of the classes which will build the instances of {@link NotificationResponse} to return.
+     */
+    private interface NotificationResponseHandler {
```
A sortir de la classe

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+    /**
+     * Build {@link PaymentResponseByNotificationResponse} instances.
+     */
+    private static class PaymentResponseByNotificationResponseHandler implements NotificationResponseHandler {
```
a sortir de la classe

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+                                    .withValue(partnerTransactionId)
+                                    .build()
+                    )
+                    .withHttpStatus(204)
```
Une petite constante pour expliciter le statut

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+                    .withTransactionId(transactionId)
+                    .withTransactionStatus(FailureTransactionStatus.builder().failureCause(e.getFailureCause()).build())
+                    .withStatusDate(new Date())
+                    .withHttpStatus(204)
```
constante

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+                    .withTransactionId(transactionId)
+                    .withTransactionStatus(FailureTransactionStatus.builder().failureCause(FailureCause.INTERNAL_ERROR).build())
+                    .withStatusDate(new Date())
+                    .withHttpStatus(204)
```
constante

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+
+    public String getTransactionIdFromURL(String url) {
+
+        int startIndex = url.indexOf(PARTNER_TRANSACTION_ID_STRING) + PARTNER_TRANSACTION_ID_STRING.length();
```
Je préfèrerais une regex, le code n'est pas simple au 1er coup d'oeil

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,179 @@
+        CustomerInformation customerInformation = CustomerInformation.Builder.aCustomerInformation()
+                .withCustomerLanguage(request.getLocale().getISO3Country())
+                .withTitle(PluginUtils.getCivilityFromPayline(request.getBuyer().getFullName().getCivility()))
+                .withFirstName(request.getBuyer().getFullName().getFirstName())
```
tu peux utiliser ton objet buyer définit un peu plus haut pour alléger le code

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,179 @@
+                .withName(request.getBuyer().getFullName().getLastName())
+                .withBirthDate(simpleDateFormat.format(request.getBuyer().getBirthday()))
+                .withEmail(request.getBuyer().getEmail())
+                .withAddressLine1(request.getBuyer().getAddressForType(Buyer.AddressType.BILLING).getStreet1())
```
pourrais tu extraire request.getBuyer().getAddressForType(Buyer.AddressType.BILLING) dans une variable pour alléger un peu le code ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/Constants.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Support for constants used everywhere in the plugin sources.
+ */
+public class Constants {
```
je sortirais les inner class et supprimerais Constants qui n'est pas très utile

JanFLORENT replied:
pourquoi pas, on le fera pour les prochains moyens de paiement

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/PluginUtils.java:
```diff
@@ -0,0 +1,281 @@
+     * @return the converted String encoded in UTF-8
+     */
+    public static String inputStreamToString(InputStream stream) {
+        BufferedReader br = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8));
```
close() le BufferedReader et InputStreamReader ou try/resources

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/PluginUtils.java:
```diff
@@ -0,0 +1,281 @@
+    public static String getGoodsCode(String cat) {
+        if (cat == null) return null;
+
+        switch (cat) {
```
sonar se plaint qu'il y a trop de case :p il faudrait transformer cela, pourquoi pas avec une map

JeremyMattio replied:
A corriger

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/http/StringResponse.java:
```diff
@@ -0,0 +1,95 @@
+public class StringResponse {
+
+    private String content;
+    private Map<String, String> headers;
```
initialiser la map ici

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/properties/AbstractProperties.java:
```diff
@@ -0,0 +1,57 @@
+            throw new PluginException("Properties filename must not be null or empty");
+        }
+
+        InputStream inputStream = AbstractProperties.class.getClassLoader().getResourceAsStream(filename);
```
close ou try/resources

pesublet commented on src/test/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationServiceTest.java:
```diff
@@ -0,0 +1,116 @@
+        PaymentFormLogo paymentFormLogo = testService.getLogo( "whatever", Locale.getDefault() );
+
+        // then:
+        assertNotNull( paymentFormLogo.getContentType() );
```
tu peux vérifier la valeur du content type

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImpl.java:
```diff
@@ -0,0 +1,177 @@
+            checkStatusRequest.getCheckStatusIn().setTransactionId("0");
+            checkStatusRequest.getCheckStatusIn().setMerchantInformation(merchantInformation);
+
+            // call in order to get the intended ErrorMessage
```
Pourrais tu détailler pourquoi tu t'attends à avoir une erreur particulière ? ce n'est pas très clair

pesublet commented on src/test/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImplTest.java:
```diff
@@ -0,0 +1,161 @@
+
+    @Test
+    void getName() {
+        String name = service.getName(Locale.FRANCE);
```
pourrais tu vérifier le contenu de name ?

pesublet commented on src/test/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImplTest.java:
```diff
@@ -0,0 +1,143 @@
+    @ParameterizedTest
+    @MethodSource("notification_set")
+    void parseNotificationChanged(String statusCode, Class responseClass) {
+        String xml = MockUtils.templateCheckStatusResponse
```
non utilisé

pesublet commented on src/test/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImplTest.java:
```diff
@@ -0,0 +1,116 @@
+        PaymentRequest request = MockUtils.aPaylinePaymentRequest();
+        InitDossierRequest initDossierRequest = service.createInitDossierFromPaymentRequest(request);
+
+        Assertions.assertNotNull(initDossierRequest);
```
il faudrait vérifier le contenu des champs en plus de la nullité

JanFLORENT replied:
C'est du simple mapping, vérifier la valeur des champs consisterait a refaire le mapping dans les tests, le risque d'erreur n'est pas réduit.

pesublet commented on src/test/java/com/payline/payment/ppewshop/service/impl/PaymentWithRedirectionServiceImplTest.java:
```diff
@@ -0,0 +1,122 @@
+        String email = "an.email@foo.bar";
+
+        PaymentResponse response = service.retrieveTransactionStatus(configuration, transactionId, email);
+        Assertions.assertEquals(responseClass, response.getClass());
```
ce serait bien de vérifier un peu plus le contenu des réponses

JeremyMattio replied:
A vérifier

pesublet commented on src/test/java/com/payline/payment/ppewshop/utils/http/HttpClientTest.java:
```diff
@@ -0,0 +1,217 @@
+        request.getCheckStatusIn().setMerchantInformation(merchantInformation);
+        request.getCheckStatusIn().setTransactionId("transactionId");
+
+        CheckStatusResponse response = client.checkStatus(configuration, request);
```
vérif de l'objet response autre que null

pesublet commented on src/test/java/com/payline/payment/ppewshop/utils/http/HttpClientTest.java:
```diff
@@ -0,0 +1,217 @@
+
+        InitDossierResponse response = client.initDossier(configuration, request);
+
+        Assertions.assertNotNull(response);
```
tester le contenu de l'objet

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/common/CheckStatusIn.java:
```diff
@@ -0,0 +1,26 @@
+
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+
+public class CheckStatusIn {
```
builder si possible

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/common/CheckStatusOut.java:
```diff
@@ -0,0 +1,46 @@
+        return creditAuthorizationNumber;
+    }
+
+    public static class StatusCode {
```
enum ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/common/InitDossierIn.java:
```diff
@@ -0,0 +1,56 @@
+
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+
+public class InitDossierIn {
```
builder si possible

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/common/MerchantOrderReference.java:
```diff
@@ -0,0 +1,54 @@
+        }
+
+        public Builder withMerchantOrderId(String merchantOrderId) {
+            this.merchantOrderId = PluginUtils.truncate(merchantOrderId, MERCHANT_ORDER_ID_LENGTH);
```
pas de manipulation de données dans le with

pesublet commented on src/main/java/com/payline/payment/ppewshop/exception/PluginException.java:
```diff
@@ -0,0 +1,103 @@
+
+    public PluginException(String message, FailureCause failureCause, Exception cause) {
+        super(message, cause);
+        if (message == null || message.length() == 0 || failureCause == null) {
```
tu peux peut etre utiliser PluginUtils.isEmpty()

pesublet commented on src/main/java/com/payline/payment/ppewshop/exception/PluginException.java:
```diff
@@ -0,0 +1,103 @@
+
+    public PluginException(String message, FailureCause failureCause) {
+        super(message);
+        if (message == null || message.length() == 0 || failureCause == null) {
```
ce "if" est fait 2 fois dans la classe, pourrais tu l'extraire dans une méthode ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImpl.java:
```diff
@@ -0,0 +1,177 @@
+    private HttpClient client = HttpClient.getInstance();
+
+
+    private static final int LENGTH = 10;
```
expliciter le nom de la variable

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImpl.java:
```diff
@@ -0,0 +1,177 @@
+            if (PluginUtils.isEmpty(infos.get(Constants.ContractConfigurationKeys.DISTRIBUTOR_NUMBER))) {
+                errors.put(Constants.ContractConfigurationKeys.DISTRIBUTOR_NUMBER
+                        , i18n.getMessage(DISTRIBUTOR_NUMBER_ERROR_EMPTY, locale));
+            } else if (infos.get(Constants.ContractConfigurationKeys.MERCHANT_CODE).length() != LENGTH) {
```
ce ne serait pas DISTRIBUTOR_NUMBER à la place de MERCHANT_CODE ?

JanFLORENT replied:
bien vu!

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/ConfigurationServiceImpl.java:
```diff
@@ -0,0 +1,177 @@
+            client.checkStatus(configuration, checkStatusRequest);
+
+        } catch (PluginException e) {
+            if (PpewShopResponseKO.ErrorCode.CODE_22002.equalsIgnoreCase(e.getErrorCode())) {
```
ça ne vaudrait pas le coup de faire des exceptions différentes au lieu d'une exception générique avec un code d'erreur ? ce n'est pas très beau les if dans le catch

JeremyMattio replied:
A traiter

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/NotificationServiceImpl.java:
```diff
@@ -0,0 +1,279 @@
+
+
+    @Override
+    public NotificationResponse parse(NotificationRequest request) {
```
découper cette méthode qui est un peu longue et complexe

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,179 @@
+
+
+    @Override
+    public PaymentResponse paymentRequest(PaymentRequest request) {
```
pourrais tu simplifier pour ne garder qu'un return ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,179 @@
+            throw new InvalidDataException("paymentRequest.buyer.address(BILLING) is required");
+        }
+
+        CustomerInformation customerInformation = CustomerInformation.Builder.aCustomerInformation()
```
attention à la nullité des champs

JeremyMattio replied:
Est on certain que les objets "fullName", "address" ou "local" ne seront jamais null ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/PluginUtils.java:
```diff
@@ -0,0 +1,281 @@
+     * @param amount the amount in cents
+     * @return Amount as String
+     */
+    public static String createStringAmount(BigInteger amount, Currency currency) {
```
Sur payline on fait en général quelque chose comme cela :
int nbDigits = currency.getDefaultFractionDigits();
final BigDecimal bigDecimal = new BigDecimal(amount);
return String.valueOf(bigDecimal.movePointLeft(nbDigits));

JanFLORENT replied:
pas mal, je prends!

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/PluginUtils.java:
```diff
@@ -0,0 +1,281 @@
+     * @param civility Payline civility to convert
+     * @return PPew civility (MME, MLE or MR)
+     */
+    public static String getCivilityFromPayline(String civility) {
```
extraire dans un business car cela s'éloigne de la méthode purement utilitaire

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/PluginUtils.java:
```diff
@@ -0,0 +1,281 @@
+     * @return
+     */
+    // todo maj le mapping quand la doc sera prete (remplacer les 000)
+    public static String getGoodsCode(String cat) {
```
extraire dans un business car cela s'éloigne de la méthode purement utilitaire

pesublet commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,179 @@
+        for (Order.OrderItem item : items) {
+            // get category and amount values
+            String cat = item.getCategory();
+            BigInteger amount = item.getAmount().getAmountInSmallestUnit()
```
je ne sais pas si cela peut arriver mais cela vaudrait le coup de verifier la nullité

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/http/StringResponse.java:
```diff
@@ -0,0 +1,95 @@
+
+            instance.headers = new HashMap<>();
+            Header[] rawHeaders = httpResponse.getAllHeaders();
+            for (int i = 0; i < rawHeaders.length; i++) {
```
for (Header header : rawHeaders) serait plus sympa

pesublet commented on src/test/java/com/payline/payment/ppewshop/MockUtils.java:
```diff
@@ -0,0 +1,460 @@
+                .anOrderItem()
+                .withReference("foo")
+                .withAmount(aPaylineAmount())
+                .withQuantity((long) 1)
```
tu peux remplacer par "1L" pour ne pas faire le cast

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/response/PpewShopResponseKO.java:
```diff
@@ -0,0 +1,100 @@
+        return cause;
+    }
+
+    public static class ErrorCode {
```
enum ?

pesublet commented on src/main/java/com/payline/payment/ppewshop/bean/response/PpewShopResponseKO.java:
```diff
@@ -0,0 +1,100 @@
+        }
+    }
+
+    public FailureCause getFailureCauseFromErrorCode() {
```
Peut être à extraire dans un business car c'est du code métier, ce n'est pas la responsabilité du bean

JeremyMattio replied:
A traiter

monext-ofrancome replied:
A traiter

pesublet commented on src/main/java/com/payline/payment/ppewshop/utils/http/HttpClient.java:
```diff
@@ -0,0 +1,218 @@
+     * @param request       request object needed to create the body
+     * @return the response body of the API call
+     */
+    public InitDossierResponse initDossier(RequestConfiguration configuration, InitDossierRequest request) {
```
c'est un peu trop métier pour être dans une classe HttpClient purement technique

JeremyMattio replied:
A déplacer

JanFLORENT committed:
Correction relecture

JanFLORENT commented:
![sonar_ppewshop_08042020](https://user-images.githubusercontent.com/41673610/78789673-bfa84300-79ad-11ea-9965-00fdf9bcf791.PNG)

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/bean/common/CheckStatusOut.java:
```diff
@@ -25,22 +25,15 @@ public MerchantOrderReference getMerchantOrderReference() {
-        public static final String I = "I";
-        public static final String R = "R";
-        public static final String C = "C";
+    public enum StatusCode {
```
Nom des attributs pas très parlant ou commentaire pour expliquer la signification de chaque statut.

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/bean/response/PpewShopResponseKO.java:
```diff
@@ -69,32 +72,40 @@ public FailureCause getFailureCauseFromErrorCode() {
 
-    public static class ErrorCode {
-        private ErrorCode() {
+    public enum ErrorCode {
```
ReturnCode pas très explicite.... A quoi correspond CODE_11001 ? Soit décrire en commentaire quel type d'erreur provoque ce 11001 soit renommer l'attribut pour que ce soit un peu plus parlant.

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/service/LogoPaymentFormConfigurationService.java:
```diff
@@ -48,23 +49,38 @@ public PaymentFormLogoResponse getPaymentFormLogo(PaymentFormLogoRequest payment
-            LOGGER.error("Unable to load file {}", filename);
-            throw new PluginException("Plugin error: unable to load the logo file");
+        String contentType = config.get("logo.contentType");
+        if (PluginUtils.isEmpty(format)){
```
contentType je pense et non format.

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -101,50 +105,57 @@ InitDossierRequest createInitDossierFromPaymentRequest(PaymentRequest request) {
         Buyer buyer = request.getBuyer();
 
-        if (buyer.getFullName() == null){
+        if (buyer.getFullName() == null) {
```
Si type String PluginUtils.isEmpty()

JanFLORENT replied:
getFullName retourne un Objet

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -176,4 +191,209 @@ String getMainCat(List<Order.OrderItem> items) {
+     * @param cat Payline category to convert
+     * @return
+     */
+    // todo maj le mapping quand la doc sera prete (remplacer les 000)
```
La doc est prête ?

martin-grescu replied:
Non, il manque encore des codes PPewShop.

JeremyMattio commented on build.gradle:
```diff
@@ -0,0 +1,132 @@
+    fatJarTest
+}
+
+dependencies {
```
le plugin doit compiler en JDK13.

JeremyMattio replied:
Le plugin compile t'il avec la JDK 13 ? Si oui tu peux resolve la discussion.

JanFLORENT replied:
c'est bon ca compile en JDK13

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,399 @@
+                .build();
+
+        MerchantOrderReference orderReference = MerchantOrderReference.Builder.aMerchantOrderReference()
+                .withMerchantOrderId(PluginUtils.truncate(request.getTransactionId(), MERCHANT_ORDER_ID_LENGTH))
```
Pk a t'on besoin de faire un truncate sur le transactionId ?

JanFLORENT replied:
c'est la Q3 de la page de questions https://docs.monext.fr/display/APMSDK/PPEWSHOP+-+Questions

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/bean/common/CheckStatusIn.java:
```diff
@@ -8,19 +8,39 @@
 
-    public MerchantInformation getMerchantInformation() {
-        return merchantInformation;
+    public CheckStatusIn(Builder builder) {
```
Pour les builder, faire plutôt un constructeur private et initialiser l'objet dans la méthode build().

JeremyMattio commented on src/main/java/com/payline/payment/ppewshop/bean/common/InitDossierIn.java:
```diff
@@ -14,43 +14,69 @@
 
-    public MerchantInformation getMerchantInformation() {
-        return merchantInformation;
+    public InitDossierIn(Builder builder) {
```
Pour les builder, faire plutôt un constructeur private et initialiser l'objet dans la méthode build().

JanFLORENT committed:
corrections relecture

Constructeurs privés
Corrections Sonar
ajout de commentaires
deplacement des methodes de HttpClient

JanFLORENT committed:
deplacement de la methode getFailureCauseFromErrorCode
Amélioration du test retrieveTransactionStatus
un seul return pour la methode getGoodsCode

JanFLORENT committed:
corrections relecture

* compatibilité JDK13
* Exceptions custom pour les messages d'erreur venant de PPEWShop
* ajout de nullCheck dans le PaymentService (paymentRequest & item)
